ramalama/model.py

```diff
@@ -171,3 +175,78 @@ def quadlet(self, model, args, exec_args):
 WantedBy=multi-user.target default.target
 """
         )
+
+    def _gen_ports(self, args):
+        if not hasattr(args, "port"):
+            return ""
+
+        ports = args.port.split(":", 2)
+        container_port = ports[0]
+        host_port = ports[0]
+        if len(ports) > 1:
+            host_port = ports[1]
+        return f"""\
+    ports:
+    - containerPort: {container_port}
+      hostPort: {host_port}"""
+
+    def _gen_volumes(self, model, args):
+        mounts = """\
+    volumeMounts:
+        - mountPath: /run/model
+          name: model"""
+
+        volumes = f"""
+  volumes:
+  - name model
+    hostPath:
+      path: {model}"""
+
+        for dev in ["dri", "kfd"]:
+            if os.path.exists("/dev/" + dev):
+                mounts = (
+                    mounts
+                    + f"""
+        - mountPath: /dev/{dev}
+          name: dri"""
+                )
+                volumes = (
+                    volumes
+                    + f""""
+  - name {dev}
+    hostPath:
+      path: /dev/{dev}"""
+                )
+
+        return mounts + volumes
+
+    def kube(self, model, args, exec_args):
```

Comment:
My experience with generating YAML files with templates has always been a poor one. There was always some indentation that got botched. I think a better solution is to create a dict and dump it using the `pyyaml` library by calling `print(yaml.dump(d))`

Reply:
Cool, do you have an example of a template for a kube.yaml?